### PR TITLE
Swap swear swindle games aron

### PR DESCRIPTION
**Pull request by cobordism**
I have started editing the swap-swear-swindle games doc.

I have made it a PR so that is is easy to see the two versions side by side. I am not asking for a merge ... yet ;)

### CODE REVIEW

File: games/main.tex [hunk 1]
```diff
@@ -134,46 +123,38 @@ \section{Introduction}
...
-anyone can send the digital cheque as part of the data in a transaction to the owner's
-chequebook account and thus trigger the transfer.
-
-Double cashing could be prevented by signing an additional field, say a serial number,
-storing it in the contract and enforcing sequential
-application. Alternatively, instead of using the payable amount, issuers sign on the cumulative
-total amount ever credited to the beneficiary. This total cumulative cashed amount is stored
-in the contract for each beneficiary. The contract ignores cheques with amount equal or less
-than the current balance, but transfer the difference if it receives a cheque with a higher
+role of the bank. When the recipient wishes to get paid, the ``cash the cheque'' by submitting it to the smart contract. The contract, after validating the signature, date and the amount specified on the cheque, transfers the amount to the beneficiary's account.
```
Comment (zelig) [inline games/main.tex]: they

File: games/main.tex [hunk 2]
```diff
@@ -134,46 +123,38 @@ \section{Introduction}
...
+by each and every node participating in the network, resulting in high transaction costs.
+One strategy to mitigate transaction costs is to defer payments and process them in bulk. 
+In exchange for reduced cost, the beneficiary must be willing to incur higher risk of settlement failure.
 
 \subsection{A simple chequebook}
 
-A stunningly simple smart contract that does exactly that was introduced in \cite{ethersphere2016sw3}
-The \gloss{chequebook} is a wallet that can process cheques issued by its owner. The cheques work
-exactly as real-world ones: the issuer signs a cheque specifying a beneficiary, a date and an amount,
+A very simple smart contract that does allows the beneficiary to choose when payments are to be processed, was introduced in \cite{ethersphere2016sw3}
```
Comment (zelig) [inline games/main.tex]: remove does

File: games/main.tex [hunk 3]
```diff
@@ -64,47 +64,36 @@
...
-Nodes organise in a network, use relaying
-to extend the scope of local interactions, and register with a deposit
-(swear) to offer service guarantees enforced by a courtroom contract
-making them accountable in case of non-compliance (swindle).
+Public proof-of-resource blockchains implement a decentralised consensus mechanism. Consensus is reached on the validity and ordering of
+transactions in a state machine. State transitions are triggered by transactions, and clients run a virtual machine to calculate these state transitions. In this process, transactions refer to specific accounts which store programs - called ``smart contracts'' - and the clients trigger functions of these programs to calculate the state transition. Limited only by the expressive power of these smart contracts, the blockchain can govern a myriad of rules of interaction, and enforce agreements.
+
+In the context of data storage and provision, we can imagine carefully designed economic incentive systems that are driven by transparent smart contracts, which regulate the peer-tp-peer data flow in such a way that the emergent properties of the network are beneficial to all of its users.
```
Comment (zelig) [inline games/main.tex]: peer-to-peer

File: games/main.tex [hunk 4]
```diff
@@ -64,47 +64,36 @@
...
+Such a system can act as a fully decentralised application platform, providing useful services while potentially
+disintermediating trusted third parties on which such services had to rely in the past.
+
+While the Turing-complete blockchain is sometimes refered to as ``the final missing piece'' of the puzzle to bring the cypherpunk vision
+of decentralisation to its completion, problems with their scalability are still a major bottleneck preventing mass adoption to real-world problems. These scalability issues include the high transaction costs for micropayments, the inherent speed and volume limitations in current blockchain designs, and are likely to remain problematic even for blockchains implementing more avanced technologies such as ``sharding'' and ``proof of stake''.
+
+In recent years, ``Side chains'' and ``state channels'' have been proposed as technologies that could remedy these shortcomings.
+While inspired by these approaches, in this paper we take a different approach. %no we don't. we're really just describing one version of a state channel, but who cares, right?
```
Comment (zelig) [inline games/main.tex]: >  %no we don't. we're really just describing one version of a state channel, but who cares, right?

probably, then let's take a definition and apply it

File: games/main.tex [hunk 5]
```diff
@@ -134,46 +123,38 @@ \section{Introduction}
...
+In exchange for reduced cost, the beneficiary must be willing to incur higher risk of settlement failure.
 
 \subsection{A simple chequebook}
 
-A stunningly simple smart contract that does exactly that was introduced in \cite{ethersphere2016sw3}
-The \gloss{chequebook} is a wallet that can process cheques issued by its owner. The cheques work
-exactly as real-world ones: the issuer signs a cheque specifying a beneficiary, a date and an amount,
+A very simple smart contract that does allows the beneficiary to choose when payments are to be processed, was introduced in \cite{ethersphere2016sw3}
+This \gloss{chequebook} contract is a wallet that can process cheques issued by its owner. The cheques work
+analgously as real-world ones: the issuer signs a cheque specifying a beneficiary, a date and an amount,
```
Comment (zelig) [inline games/main.tex]: > analogously as

doesn't sound right to me

File: games/main.tex [hunk 6]
```diff
@@ -134,46 +123,38 @@ \section{Introduction}
...
-total amount ever credited to the beneficiary. This total cumulative cashed amount is stored
-in the contract for each beneficiary. The contract ignores cheques with amount equal or less
-than the current balance, but transfer the difference if it receives a cheque with a higher
+role of the bank. When the recipient wishes to get paid, the ``cash the cheque'' by submitting it to the smart contract. The contract, after validating the signature, date and the amount specified on the cheque, transfers the amount to the beneficiary's account.
+%Analogously to the person taking the cheque to the bank to cash it, anyone can send the digital cheque as part of the data in a transaction to the owner's chequebook account and thus trigger the transfer.
+
+Since these digital cheques are copyable files, care must be taken that the same cheque cannot be casehd twice. Such ``double cashing'' can be prevented by giving each cheque given to a particular beneficiary a serial number which the contract will store when the cheque is cashed. Perhaps the chequebook contract would enforce cashing of cheques in sequential order, thus needing to store only a single serial number per beneficiary.
```
Comment (zelig) [inline games/main.tex]: copyable?

casehd -> cashed

File: games/main.tex [hunk 7]
```diff
@@ -134,46 +123,38 @@ \section{Introduction}
...
-cheques constitutes a risk to the beneficiary.
+
+This simple trick also makes it possible to cash cheques in bulk, because only the current `last cheque' must ever be processed. This achieves the reduction of transaction costs alluded to above.
+
+Incidentally, the cumulative amount stored in the contract, represents the total of all outgoing payments that have been honoured and thus it serves as a credit history for the owner.
+
+The amount deposited in the chequebook (\gloss{global balance}) serves as collateral for the cheques. It is pooled over the beneficiaries of all outstanding cheques.
+In this simplest form the chequebook has the same guarantee as real-world cheques: none. When a chequebook contract has zero balance, and cheque that is submitted will bounce. Since funds can be freely moved out of the chequebook wallet at any time, solvency at the time of cashing can never be guaranteed. This is the trade off between transaction costs and risk of settlement failure.
+
+% Parts of this deposit can be locked (\gloss{global deposit}).
```
Comment (zelig) [inline games/main.tex:150]: we skip this?

File: games/main.tex [hunk 8]
```diff
@@ -293,34 +268,30 @@ \subsection{SWAP accounting with chequebook}
...
-Now the balance tilts in A's favour but the cheques from A to B have not been cashed.
-In this scenario, instead of sending a cheque to A, B can waive part of their entitlement.
-This justifies swap as \emph{send waiver as payment} (see figure \ref{fig:waiverswap}).
-A waiver essentially implements a proof that a cheque is destroyed, i.e., a promise
-that (part of) an uncashed cheque will not be redeemed.
-
-A \gloss{waiver} is implemented as a note signed by the peer holding uncashed cheques.
-The note specifies the current \gloss{swap index} (sequential number) and
-indicates the amount of dept they agree to waive from their entitlement.
+If the imbalance in the swap channel results from high variance as opposeed to unequal consumption, it will happen that, after a period of accumulating cheques the channel balance starts tilting the other way. Nominally it is now up to the other party to issue cheques to its peer resulting an uncashed cheques accumulating on both sides.
```
Comment (zelig) [inline games/main.tex]: resulting in

File: games/main.tex [hunk 9]
```diff
@@ -293,34 +268,30 @@ \subsection{SWAP accounting with chequebook}
...
+
+\subsubsection*{Cashing waivers}
+Analogously to cheques, waivers are accepted by the contract. %if sent in as data in a transaction together with the issuer's address.
+Upon receiving a waiver the contract verifies the data, and %signature and checks if the swap index matches the one in the waiver.
+if the waiver is valid, the peer's cumulative cashed balance is increased by the waived amount pretending the waived amount was paid out. Then the new cumulative cashed balance is compared to the one recorded when the cheque was received. Any remaining positive balance is transfered to the beneficiary, the cumulative sum is cleared from storage and the swap index is incremented.
+
+%%I stop here for now because the use of 'the swap index'  is confusing and I will rewrite it later ot be more clear.... a sequential index of waivers, stored by the checkbook separately from the serial number of the cheques but functioning like... how should this work? unclearly written
```
Comment (zelig) [inline games/main.tex:293]: actually, see fabio's comment, it is to be called swap cycle index, and is not indexing the waiver, just punctuates the periods when withheld cheques are used. 
Yes it is a little unclearly written here.

File: games/main.tex (hunk 4 above)
Comment (cobordism) [reply games/main.tex]: How about
```
In recent years, ``Side chains'' and ``state channels'' have been proposed as technologies that could remedy these shortcomings, and they directly inspired some of the approaches taken in this paper.
```

File: games/main.tex (hunk 6 above)
Comment (cobordism) [reply games/main.tex]: fixed, reformulated.

File: games/main.tex (hunk 7 above)
Comment (cobordism) [reply games/main.tex:150]: unless you can explain it to me and say why it should stay there... For now I say we can do without it.

File: games/main.tex (hunk 8 above)
Comment (homotopycolimit) [reply games/main.tex]: is the result of

**Pull request closed without merging**